app/src/main/java/com/allan/boardbuddies/viewmodels/EditNoteViewModel.java

```diff
@@ -0,0 +1,40 @@
+package com.allan.boardbuddies.viewmodels;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.ViewModel;
+
+import com.allan.boardbuddies.models.Note;
+import com.allan.boardbuddies.repositories.NoteRepository;
+
+public class EditNoteViewModel extends ViewModel {
+    private NoteRepository noteRepository;
+
+    private Note localNote;
+
+    public EditNoteViewModel(){
+        noteRepository = NoteRepository.getInstance();
```

Comment:
why don't you call initRepository here instead of in the mainactivity? this way we only load the notes when we need it

Reply:
idk if its thread safe this way also i don't have app context in viewmodel